src/main/java/com/iota/iri/controllers/TransactionViewModel.java

```diff
@@ -976,66 +975,45 @@ public String toString() {
     private static void cachePut(Tangle tangle, TransactionViewModel transactionViewModel, Hash hash) throws Exception {
         Cache<Indexable, TransactionViewModel> cache = tangle.getCache(TransactionViewModel.class);
         if (cache.getSize() >= cache.getConfiguration().getMaxSize()) {
```

Comment:
Maybe this should be changed into `while` on all implementations? Just to make this more robust

Reply:
I don't follow

Reply:
All the notification from the migration tool made me miss this...

Imagine that due to some weird scenario we have `cache.getSize() == cache.getConfiguration().getMaxSize() + 100`

I don't see how this can happen but still, code changes

So if you put a while you will solve this :-)?

You can also argue back that this shouldn't happen, and we're just attempting to mask bugs (which is also a valid statement)

Reply:
I dont see how this can happen gal,Could you describe this scenario? And if it happens, we should have a unit test for it :)

Reply:
currently it can't happen
the point is that code changes

What the code does now is not what it will do tomorrow

Reply:
Is it not better to throw an exception in `cache.put` if you attempt to put when it's full?

Reply:
this will indeed make the problem easier to spot!!!
So yes

Reply:
But then we'll be double-checking size constraints for each cache put. A check in `TVM.cachePut` and check again in `cache.put`.
I think we should leave it as it is. :)

Reply:
hmm... I don't think those checks are so expensive though

Reply:
We’ve come back to this:

https://iotafoundation.slack.com/archives/GRVHJAA9F/p1580815146089200
> so I added another cache unit test: cacheSizeShouldNotGrowPastMaxSize and it’s failing.
Because if cache is full and then a get is called for a recently released item that’s still in weakstore, it puts it back in strongstore.
So there’s a +1 on max size there. :disappointed:
We cannot release in cache.put because we have to make sure its released to DB if necessary (from the VM)

so now I’m getting some cacheExceptions from unit tests due to the above
Since we are now releasing 1 at a time, it means cache is always full.
So if cache.get is called on an item that is in weakstore, it’ll try to add it to already full cache which will throw Exception.

What I think we should do, for now, is to use the `while` option @GalRogozinski  proposed.
Then open an issue for refactoring that can be looked at a later time